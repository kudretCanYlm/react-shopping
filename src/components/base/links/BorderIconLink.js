import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { EMPTY_ICON } from './common/common-props';

const BorderIconLink = ({ children, to, className }) => {
  return (
    <div className={`border-icon-link flex-row link-base ${className}`}>
      <Link className="link" to={to}>
        {children}
      </Link>
    </div>
  );
};

BorderIconLink.propTypes = {
  children: PropTypes.object.isRequired,
  to: PropTypes.string.isRequired,
  className: PropTypes.string
};

BorderIconLink.defaultProps = {
  children: EMPTY_ICON,
  to: '/',
  className: ''
};

export default BorderIconLink;
