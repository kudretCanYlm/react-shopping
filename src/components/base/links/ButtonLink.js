import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const ButtonLink = ({ children, to, className }) => {
  return (
    <Link className={`button-link ${className}`} to={to}>
      {children}
    </Link>
  );
};

//proptypes
ButtonLink.propTypes = {
  children: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  className: PropTypes.string
};

//default props
ButtonLink.defaultProps = {
  to: '',
  className: ''
};

export default ButtonLink;
