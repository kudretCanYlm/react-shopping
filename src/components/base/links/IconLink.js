import React from 'react';
import { Link } from 'react-router-dom';
import '../../../styles/base/base.css';

const IconLink = ({ children, text, to = '/' }) => {
  return (
    <div className="link-base">
      <Link className="link link-icon flex-row" to={to}>
        <div className="icon">{children}</div>
        <h2>{text}</h2>
      </Link>
    </div>
  );
};

export default IconLink;
