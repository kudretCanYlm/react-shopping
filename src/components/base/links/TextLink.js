import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const TextLink = ({ children, to, className }) => {
  return (
    <Link className={`link-text ${className}`} to={to}>
      {children}
    </Link>
  );
};

//proptypes
TextLink.propTypes = {
  children: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  className: PropTypes.string
};

//default props
TextLink.defaultProps = {
  children: 'empty',
  to: '/',
  className: ''
};

export default TextLink;
