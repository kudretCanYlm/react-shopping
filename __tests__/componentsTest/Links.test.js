import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import '@testing-library/jest-dom';
import { createMemoryHistory } from 'history';
//components
import BlankLink from 'components/base/links/Blanklink';
import { MemoryRouter, Router } from 'react-router-dom';
import BorderIconLink from 'components/base/links/BorderIconLink';
import { DiAtom, DiApple } from 'react-icons/di';
import ButtonLink from 'components/base/links/ButtonLink';
import IconLink from 'components/base/links/IconLink';
import TextLink from 'components/base/links/TextLink';

Enzyme.configure({ adapter: new Adapter() });

//variables
const blankHref = 'www.youtube.com/';
const longBlankHref = 'https://www.npmjs.com/package/webpack-bundle-analyzer';
const linkClassName = 'test classname';
const history = createMemoryHistory({ initialEntries: ['/home'] });
const to_1 = '/test';
const to_2 = '/main';
const icon = <DiAtom fontSize={20} color="red" />;
const notContainIcon = <DiApple fontSize={30} color="red" />;
const linkText = 'Click Here';
const notContainButtonText = "Don't Click Here";

describe('BlankLinkTest', () => {
  it('render correctly', () => {
    const component = renderer
      .create(<BlankLink className={linkClassName} href={blankHref} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('it has my className,href,href text', () => {
    const linkCom = shallow(<BlankLink className={linkClassName} href={blankHref} />);

    expect(linkCom.hasClass(linkClassName)).toEqual(true);
    expect(linkCom.html()).toContain(`href=\"https://${blankHref}\"`);
    expect(linkCom.html()).toContain(blankHref);

    expect(linkCom.hasClass('not include ClassName')).toEqual(false);
    expect(linkCom.html()).not.toContain(`href=\"${blankHref}\"`);
    expect(linkCom.html()).not.toContain('https://www.npmjs.com/package/webpack-bundle-analyzer');
  });

  it("it hasn't href text length more 20", () => {
    const linkCom = shallow(<BlankLink className={linkClassName} href={longBlankHref} />);

    expect(linkCom.children(0).contains(longBlankHref.substring(0, 20))).toEqual(true);
    expect(linkCom.children(0).contains(longBlankHref)).toEqual(false);
  });
});

describe('BorderIconLinkTest', () => {
  it('render correctly', () => {
    const component = renderer.create(
      <MemoryRouter history={history}>
        <BorderIconLink className={linkClassName} to={to_1}>
          {icon}
        </BorderIconLink>
      </MemoryRouter>
    );

    expect(component).toMatchSnapshot();
  });

  it('it has my children,route and className', () => {
    const linkCom = shallow(
      <MemoryRouter history={history}>
        <BorderIconLink className={linkClassName} to={to_1}>
          {icon}
        </BorderIconLink>
      </MemoryRouter>
    );

    expect(linkCom.contains(icon)).toEqual(true);
    expect(linkCom.html()).toContain(`href=\"${to_1}\"`);
    expect(linkCom.children().hasClass(linkClassName)).toEqual(true);

    expect(linkCom.contains(notContainIcon)).toEqual(false);
    expect(linkCom.html()).not.toContain(`href=\"${to_2}\"`);
    expect(linkCom.children().hasClass('not include ClassName')).toEqual(false);
  });
});

describe('ButtonLinkTest', () => {
  it('render correctly', () => {
    const component = renderer
      .create(
        <MemoryRouter history={history}>
          <ButtonLink className={linkClassName} to={to_1}>
            {linkText}
          </ButtonLink>
        </MemoryRouter>
      )
      .toJSON();

    expect(component).toMatchSnapshot(component);
  });

  it('it has my children,route and className', () => {
    const linkCom = shallow(
      <MemoryRouter history={history}>
        <ButtonLink className={linkClassName} to={to_1}>
          {linkText}
        </ButtonLink>
      </MemoryRouter>
    );

    expect(linkCom.contains(linkText)).toEqual(true);
    expect(linkCom.html()).toContain(`href=\"${to_1}\"`);
    expect(linkCom.children().hasClass(linkClassName)).toEqual(true);

    expect(linkCom.contains(notContainButtonText)).toEqual(false);
    expect(linkCom.html()).not.toContain(`href=\"${to_2}\"`);
    expect(linkCom.children().hasClass('not include ClassName')).toEqual(false);
  });
});

describe('IconLinkTest', () => {
  it('render correctly', () => {
    const component = renderer
      .create(
        <MemoryRouter history={history}>
          <IconLink text={linkText} to={to_1} className={linkClassName}>
            {icon}
          </IconLink>
        </MemoryRouter>
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('it has my children ,text ,route and className', () => {
    const linkCom = shallow(
      <MemoryRouter history={history}>
        <IconLink text={linkText} to={to_1} className={linkClassName}>
          {icon}
        </IconLink>
      </MemoryRouter>
    );

    expect(linkCom.contains(icon)).toEqual(true);
    expect(linkCom.html()).toContain(linkText);
    expect(linkCom.html()).toContain(`href=\"${to_1}\"`);
    expect(linkCom.children().hasClass(linkClassName)).toEqual(true);

    expect(linkCom.contains(notContainIcon)).toEqual(false);
    expect(linkCom.html()).not.toContain(notContainButtonText);
    expect(linkCom.html()).not.toContain(`href=\"${to_2}\"`);
    expect(linkCom.children().hasClass('not include ClassName')).toEqual(false);
  });
});

describe('TextLinkTest', () => {

  it("render correctly", () => {
    const component = renderer.create(
      <MemoryRouter history={history}>
        <TextLink className={linkClassName} to={to_1}>
          {linkText}
        </TextLink>
      </MemoryRouter>
    )
  })

  it("it has my children,route and className", () => {

    const linkCom = shallow(
      <MemoryRouter history={history}>
        <TextLink className={linkClassName} to={to_1}>
          {linkText}
        </TextLink>
      </MemoryRouter>
    )

    expect(linkCom.html()).toContain(linkText);
    expect(linkCom.html()).toContain(`href=\"${to_1}\"`);
    expect(linkCom.children().hasClass(linkClassName)).toEqual(true);

    expect(linkCom.html()).not.toContain(notContainButtonText);
    expect(linkCom.html()).not.toContain(`href=\"${to_2}\"`);
    expect(linkCom.children().hasClass('not include ClassName')).toEqual(false);

  })
});
