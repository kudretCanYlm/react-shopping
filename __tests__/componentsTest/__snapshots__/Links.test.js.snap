// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlankLinkTest render correctly 1`] = `
<div
  className="blank-link flex-center-items flex-row test classname"
>
  <svg
    className="icon-margin-right-1"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 1024 1024"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M574 665.4a8.03 8.03 0 0 0-11.3 0L446.5 781.6c-53.8 53.8-144.6 59.5-204 0-59.5-59.5-53.8-150.2 0-204l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3l-39.8-39.8a8.03 8.03 0 0 0-11.3 0L191.4 526.5c-84.6 84.6-84.6 221.5 0 306s221.5 84.6 306 0l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3L574 665.4zm258.6-474c-84.6-84.6-221.5-84.6-306 0L410.3 307.6a8.03 8.03 0 0 0 0 11.3l39.7 39.7c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c53.8-53.8 144.6-59.5 204 0 59.5 59.5 53.8 150.2 0 204L665.3 562.6a8.03 8.03 0 0 0 0 11.3l39.8 39.8c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c84.5-84.6 84.5-221.5 0-306.1zM610.1 372.3a8.03 8.03 0 0 0-11.3 0L372.3 598.7a8.03 8.03 0 0 0 0 11.3l39.6 39.6c3.1 3.1 8.2 3.1 11.3 0l226.4-226.4c3.1-3.1 3.1-8.2 0-11.3l-39.5-39.6z"
    />
  </svg>
  <a
    href="https://www.youtube.com/"
    target="blank"
  >
    www.youtube.com/
  </a>
</div>
`;

exports[`BorderIconLinkTest render correctly 1`] = `
<div
  className="border-icon-link flex-row link-base test classname"
>
  <a
    className="link"
    href="/test"
    onClick={[Function]}
  >
    <svg
      color="red"
      fill="currentColor"
      fontSize={20}
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "red",
        }
      }
      version="1.1"
      viewBox="0 0 32 32"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.799 4.652c-1.485 0.324-2.271 2.045-2.104 4.593 0.051 0.738 0.043 0.666 0.196 1.609 0.064 0.38 0.107 0.7 0.098 0.709-0.008 0.013-0.269 0.077-0.572 0.149-2.019 0.465-3.505 1.165-4.397 2.070-0.602 0.606-0.854 1.17-0.845 1.882 0.004 0.401 0.137 0.841 0.38 1.264 0.209 0.363 0.956 1.101 1.447 1.434 1.029 0.692 1.345 0.79 1.626 0.508 0.12-0.119 0.145-0.179 0.145-0.32 0-0.273-0.094-0.405-0.414-0.581-1.409-0.781-2.147-1.592-2.147-2.369 0-0.282 0.098-0.538 0.333-0.845 0.619-0.824 2.113-1.562 4.115-2.036 0.529-0.124 0.632-0.132 0.632-0.043 0 0.115 0.427 1.481 0.7 2.228l0.273 0.751-0.337 0.645c-0.184 0.354-0.448 0.892-0.585 1.2-1.959 4.316-2.284 7.743-0.867 9.152 0.333 0.333 0.606 0.487 1.054 0.602 1.033 0.265 2.399-0.132 3.931-1.144 0.534-0.354 0.653-0.487 0.653-0.721 0-0.282-0.307-0.555-0.581-0.512-0.077 0.013-0.376 0.179-0.662 0.367-0.632 0.422-1.34 0.773-1.853 0.926-0.525 0.154-1.093 0.162-1.417 0.021-0.995-0.44-1.225-2.215-0.606-4.678 0.29-1.17 0.956-2.928 1.558-4.128l0.239-0.482 0.132 0.299c0.248 0.572 1.212 2.437 1.588 3.073 2.079 3.534 4.422 6.125 6.501 7.184 1.473 0.751 2.689 0.683 3.517-0.201 0.61-0.645 0.909-1.584 0.96-2.992 0.081-2.425-0.709-5.579-2.254-8.96-0.205-0.453-0.41-0.862-0.448-0.905-0.094-0.102-0.333-0.171-0.495-0.137s-0.359 0.231-0.388 0.397c-0.034 0.158 0.004 0.265 0.384 1.088 1.059 2.284 1.801 4.683 2.087 6.744 0.094 0.679 0.111 2.151 0.026 2.604-0.085 0.457-0.252 0.931-0.431 1.204-0.286 0.44-0.615 0.619-1.157 0.615-1.609-0.004-4.145-2.215-6.399-5.571-1.037-1.55-1.993-3.3-2.732-5.011l-0.265-0.61 0.371-0.627c0.478-0.811 0.982-1.579 1.545-2.369l0.448-0.627h0.692c4.747 0 9.459 1.076 11.867 2.702 0.551 0.371 1.080 0.914 1.264 1.289 0.128 0.265 0.145 0.337 0.145 0.64-0.004 0.286-0.021 0.376-0.119 0.563-0.294 0.572-1.042 1.14-2.079 1.592-0.487 0.209-0.64 0.354-0.64 0.602 0 0.23 0.094 0.397 0.273 0.482 0.196 0.094 0.265 0.085 0.581-0.043 1.49-0.602 2.565-1.49 2.903-2.395 0.623-1.665-0.683-3.347-3.564-4.602-2.518-1.101-6.219-1.789-10.070-1.87l-0.423-0.009 0.482-0.555c0.555-0.645 1.78-1.87 2.305-2.309 1.246-1.050 2.361-1.716 3.321-1.989 0.474-0.137 1.059-0.132 1.362 0.004 0.41 0.184 0.696 0.598 0.854 1.238 0.098 0.388 0.098 1.575 0 2.147-0.111 0.632-0.098 0.743 0.073 0.913 0.124 0.124 0.175 0.145 0.354 0.145 0.38 0 0.478-0.141 0.593-0.832 0.060-0.354 0.081-0.692 0.081-1.387 0-0.811-0.013-0.965-0.098-1.302-0.269-1.063-0.926-1.797-1.806-2.006-2.040-0.478-5.161 1.485-8.264 5.208-0.256 0.303-0.495 0.602-0.534 0.653-0.064 0.094-0.107 0.102-0.726 0.141-0.359 0.021-1.016 0.081-1.464 0.132-1.187 0.137-1.093 0.149-1.161-0.158-0.179-0.858-0.239-1.46-0.243-2.39-0.004-1.007 0.030-1.306 0.213-1.865 0.196-0.593 0.529-0.995 0.952-1.135 0.205-0.073 0.709-0.064 1.007 0.013 0.499 0.132 1.204 0.508 1.844 0.99 0.38 0.286 0.512 0.337 0.713 0.269 0.23-0.073 0.367-0.265 0.367-0.504 0-0.179-0.017-0.213-0.205-0.393-0.265-0.256-1.033-0.768-1.498-0.999-0.879-0.44-1.648-0.581-2.339-0.431zM12.4 12.216c-0.004 0.021-0.282 0.44-0.61 0.935s-0.653 0.995-0.721 1.11l-0.124 0.209-0.102-0.277c-0.128-0.337-0.525-1.643-0.525-1.725 0-0.077 0.188-0.107 1.579-0.252 0.29-0.030 0.521-0.030 0.504 0zM15.649 14.854c-0.303 0.098-0.598 0.316-0.773 0.576-0.525 0.773-0.269 1.78 0.555 2.185 0.256 0.128 0.32 0.141 0.67 0.141s0.414-0.013 0.67-0.141c1.114-0.546 1.089-2.168-0.043-2.689-0.299-0.137-0.781-0.166-1.080-0.073z"
      />
    </svg>
  </a>
</div>
`;

exports[`ButtonLinkTest render correctly 1`] = `
Object {
  "children": Array [
    "Click Here",
  ],
  "props": Object {
    "className": "button-link test classname",
    "href": "/test",
    "onClick": [Function],
    "target": undefined,
  },
  "type": "a",
}
`;

exports[`IconLinkTest render correctly 1`] = `
<div
  className="link-base test classname"
>
  <a
    className="link link-icon flex-row"
    href="/test"
    onClick={[Function]}
  >
    <div
      className="icon"
    >
      <svg
        color="red"
        fill="currentColor"
        fontSize={20}
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "red",
          }
        }
        version="1.1"
        viewBox="0 0 32 32"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.799 4.652c-1.485 0.324-2.271 2.045-2.104 4.593 0.051 0.738 0.043 0.666 0.196 1.609 0.064 0.38 0.107 0.7 0.098 0.709-0.008 0.013-0.269 0.077-0.572 0.149-2.019 0.465-3.505 1.165-4.397 2.070-0.602 0.606-0.854 1.17-0.845 1.882 0.004 0.401 0.137 0.841 0.38 1.264 0.209 0.363 0.956 1.101 1.447 1.434 1.029 0.692 1.345 0.79 1.626 0.508 0.12-0.119 0.145-0.179 0.145-0.32 0-0.273-0.094-0.405-0.414-0.581-1.409-0.781-2.147-1.592-2.147-2.369 0-0.282 0.098-0.538 0.333-0.845 0.619-0.824 2.113-1.562 4.115-2.036 0.529-0.124 0.632-0.132 0.632-0.043 0 0.115 0.427 1.481 0.7 2.228l0.273 0.751-0.337 0.645c-0.184 0.354-0.448 0.892-0.585 1.2-1.959 4.316-2.284 7.743-0.867 9.152 0.333 0.333 0.606 0.487 1.054 0.602 1.033 0.265 2.399-0.132 3.931-1.144 0.534-0.354 0.653-0.487 0.653-0.721 0-0.282-0.307-0.555-0.581-0.512-0.077 0.013-0.376 0.179-0.662 0.367-0.632 0.422-1.34 0.773-1.853 0.926-0.525 0.154-1.093 0.162-1.417 0.021-0.995-0.44-1.225-2.215-0.606-4.678 0.29-1.17 0.956-2.928 1.558-4.128l0.239-0.482 0.132 0.299c0.248 0.572 1.212 2.437 1.588 3.073 2.079 3.534 4.422 6.125 6.501 7.184 1.473 0.751 2.689 0.683 3.517-0.201 0.61-0.645 0.909-1.584 0.96-2.992 0.081-2.425-0.709-5.579-2.254-8.96-0.205-0.453-0.41-0.862-0.448-0.905-0.094-0.102-0.333-0.171-0.495-0.137s-0.359 0.231-0.388 0.397c-0.034 0.158 0.004 0.265 0.384 1.088 1.059 2.284 1.801 4.683 2.087 6.744 0.094 0.679 0.111 2.151 0.026 2.604-0.085 0.457-0.252 0.931-0.431 1.204-0.286 0.44-0.615 0.619-1.157 0.615-1.609-0.004-4.145-2.215-6.399-5.571-1.037-1.55-1.993-3.3-2.732-5.011l-0.265-0.61 0.371-0.627c0.478-0.811 0.982-1.579 1.545-2.369l0.448-0.627h0.692c4.747 0 9.459 1.076 11.867 2.702 0.551 0.371 1.080 0.914 1.264 1.289 0.128 0.265 0.145 0.337 0.145 0.64-0.004 0.286-0.021 0.376-0.119 0.563-0.294 0.572-1.042 1.14-2.079 1.592-0.487 0.209-0.64 0.354-0.64 0.602 0 0.23 0.094 0.397 0.273 0.482 0.196 0.094 0.265 0.085 0.581-0.043 1.49-0.602 2.565-1.49 2.903-2.395 0.623-1.665-0.683-3.347-3.564-4.602-2.518-1.101-6.219-1.789-10.070-1.87l-0.423-0.009 0.482-0.555c0.555-0.645 1.78-1.87 2.305-2.309 1.246-1.050 2.361-1.716 3.321-1.989 0.474-0.137 1.059-0.132 1.362 0.004 0.41 0.184 0.696 0.598 0.854 1.238 0.098 0.388 0.098 1.575 0 2.147-0.111 0.632-0.098 0.743 0.073 0.913 0.124 0.124 0.175 0.145 0.354 0.145 0.38 0 0.478-0.141 0.593-0.832 0.060-0.354 0.081-0.692 0.081-1.387 0-0.811-0.013-0.965-0.098-1.302-0.269-1.063-0.926-1.797-1.806-2.006-2.040-0.478-5.161 1.485-8.264 5.208-0.256 0.303-0.495 0.602-0.534 0.653-0.064 0.094-0.107 0.102-0.726 0.141-0.359 0.021-1.016 0.081-1.464 0.132-1.187 0.137-1.093 0.149-1.161-0.158-0.179-0.858-0.239-1.46-0.243-2.39-0.004-1.007 0.030-1.306 0.213-1.865 0.196-0.593 0.529-0.995 0.952-1.135 0.205-0.073 0.709-0.064 1.007 0.013 0.499 0.132 1.204 0.508 1.844 0.99 0.38 0.286 0.512 0.337 0.713 0.269 0.23-0.073 0.367-0.265 0.367-0.504 0-0.179-0.017-0.213-0.205-0.393-0.265-0.256-1.033-0.768-1.498-0.999-0.879-0.44-1.648-0.581-2.339-0.431zM12.4 12.216c-0.004 0.021-0.282 0.44-0.61 0.935s-0.653 0.995-0.721 1.11l-0.124 0.209-0.102-0.277c-0.128-0.337-0.525-1.643-0.525-1.725 0-0.077 0.188-0.107 1.579-0.252 0.29-0.030 0.521-0.030 0.504 0zM15.649 14.854c-0.303 0.098-0.598 0.316-0.773 0.576-0.525 0.773-0.269 1.78 0.555 2.185 0.256 0.128 0.32 0.141 0.67 0.141s0.414-0.013 0.67-0.141c1.114-0.546 1.089-2.168-0.043-2.689-0.299-0.137-0.781-0.166-1.080-0.073z"
        />
      </svg>
    </div>
    <h2>
      Click Here
    </h2>
  </a>
</div>
`;
